Python/pywarpx/_libwarpx.py

```diff
@@ -392,14 +392,14 @@ def get_nattr_species(self, species_name):
 
         '''
         return self.libwarpx_so.warpx_nCompsSpecies(
-            ctypes.c_char_p(species_name.encode('utf-8')))
+            ctypes.c_char_p(species_name.encode('ascii')))
 
     def amrex_init(self, argv, mpi_comm=None):
         # --- Construct the ctype list of strings to pass in
         argc = len(argv)
         argvC = (_LP_c_char * (argc+1))()
```

Comment:
Hm.... maybe we have an off-by-one error. I think this should not read
```suggestion
        argvC = (_LP_c_char * argc)()
```
https://stackoverflow.com/a/62422953/2719194

We already have that argument in there when we pass it in this function. For the `Examples/Physics_applications/laser_acceleration/PICMI_inputs_3d.py` example its:
```
amrex_init ['warpx', 'warpx.verbose = 1', 'warpx.use_filter = 1', 'warpx.serialize_ics = 1', 'warpx.do_dynamic_scheduling = 0', 'warpx.do_moving_window = 1', 'warpx.moving_window_dir = z', 'warpx.moving_window_v = 1.0', 'warpx.cfl = 1.0', 'warpx.do_dive_cleaning = 0', 'amr.n_cell = 32 32 256', 'amr.max_grid_size = 64', 'amr.blocking_factor = 32', 'amr.max_level = 0', 'geometry.dims = 3', 'geometry.prob_lo = -3e-05 -3e-05 -5.6e-05', 'geometry.prob_hi = 3e-05 3e-05 1.2e-05', 'boundary.field_lo = periodic periodic pec', 'boundary.field_hi = periodic periodic pec', 'boundary.particle_lo = periodic periodic absorbing', 'boundary.particle_hi = periodic periodic absorbing', 'algo.particle_shape = 3', 'algo.maxwell_solver = Yee', 'particles.species_names = electrons', 'electrons.mass = m_e', 'electrons.charge = -q_e', 'electrons.injection_style = nuniformpercell', 'electrons.initialize_self_fields = 0', 'electrons.num_particles_per_cell_each_dim = 1 1 1', 'electrons.xmin = -2e-05', 'electrons.xmax = 2e-05', 'electrons.ymin = -2e-05', 'electrons.ymax = 2e-05', 'electrons.zmin = 0', 'electrons.profile = constant', 'electrons.density = 2e+23', 'electrons.momentum_distribution_type = constant', 'electrons.ux = 0.0', 'electrons.uy = 0.0', 'electrons.uz = 0.0', 'electrons.do_continuous_injection = 1', 'lasers.names = laser1', 'laser1.profile = Gaussian', 'laser1.wavelength = 8e-07', 'laser1.e_max = 16000000000000.0', 'laser1.polarization = 0 1 0', 'laser1.profile_waist = 5e-06', 'laser1.profile_duration = 1.5e-14', 'laser1.do_continuous_injection = 0', 'laser1.position = 0.0 0.0 9e-06', 'laser1.direction = 0 0 1', 'laser1.profile_focal_distance = 0.0001', 'laser1.profile_t_peak = 3e-14', 'diagnostics.diags_names = diag1', 'diag1.diag_type = Full', 'diag1.format = plotfile', 'diag1.intervals = 100', 'diag1.fields_to_plot = Bx By Bz Ex Ey Ez jx jy jz rho', 'diag1.file_prefix = ./Python_LaserAcceleration_plt']
```

Reply:
Yes, I think you are correct, this should be `argvC = (_LP_c_char * argc)()`.